### PR TITLE
Small edits to tensorflow examples in the earlier chapters.

### PR DESCRIPTION
**Pull request by tomerk**
These include:
* Switching tf.constant(range()) to tf.range
* Changing an incorrect note that tensorflow doesn't support `len`
* removing unnecessary `persistent=True` in gradienttapes that don't compute a gradient more than once.
* replacing a `sum` with a `tf.reduce_sum` for consistency (and more performant)

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1069/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1069/

**Comment by mli** [comment]
@astonzhang is the CI error related to the d2l plotting change?

**Comment by astonzhang** [comment]
> @astonzhang is the CI error related to the d2l plotting change?

Fixing it now.

**Comment by mli** [comment]
Job d2l-en/PR-1069/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1069/
